packages/react/src/components/DataTable/DataTable.tsx

```diff
@@ -474,7 +476,7 @@ class DataTable<RowType, ColTypes extends any[]> extends React.Component<
       ...rest,
       key: header.key,
       sortDirection,
-      isSortable,
+      isSortable: header.isSortable || isSortable, // header check first, since every header defaults to the table isSortable prop
```

Comment:
I think the nullish coalescing operator you had prior to https://github.com/carbon-design-system/carbon/commit/f34935f7fb9a2930a273eecf39b4ad90f4a66b44 was correct. Otherwise, if someone sets `header.isSortable` to `false` and `isSortable` is `true`, the expression `false || true` will evaluate to `true`, which I don't think is optimal. `header.isSortable` should take precedence over `isSortable`.

Also, should this scenario be tested?